quill-sql/src/main/scala/io/getquill/sources/sql/SqlSource.scala

```diff
@@ -117,35 +118,35 @@ abstract class SqlSource[D <: SqlIdiom, N <: NamingStrategy, R: ClassTag, S: Cla
 
   def run[T](
     quoted: Quoted[T]
-  ): QueryResult[T] = macro SqlSourceMacro.run[R, S]
+  ): SingleQueryResult[T] = macro SqlSourceMacro.run[R, S]
```

Comment:
This could use `SqlSourceMacro.runSingle[R, S]`, avoiding the encoding of `isQuery` with the quotation.